pkg/query/querier.go

```diff
@@ -260,6 +265,18 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 		return nil, errors.Wrap(err, "convert matchers")
 	}
 
+	tenantSpecified := false
+	for _, sms := range sms {
```

Comment:
This is security impacting code, so it would be best to put it into a helper or even import a production-grade promllabel proxy library for this work (and test properly).